rust/src/ssh/ssh.rs

```diff
@@ -156,20 +160,31 @@ impl SSHState {
             match parser::ssh_parse_record(input) {
                 Ok((rem, head)) => {
                     SCLogDebug!("SSH valid record {}", head);
-                    input = rem;
-                    if head.msg_code == SSH_MSG_NEWKEYS {
-                        hdr.flags = SSHConnectionState::SshStateFinished;
-                        if ohdr.flags >= SSHConnectionState::SshStateFinished {
-                            unsafe {
-                                AppLayerParserStateSetFlag(
-                                    pstate,
-                                    APP_LAYER_PARSER_NO_INSPECTION
+                    match head.msg_code {
+                        parser::MessageCode::SshMsgKexinit if unsafe {SSHHasshIsEnabled()} && !input.is_empty() => {
+                        	match parser::parse_packet_key_exchange(&input[SSH_RECORD_HEADER_LEN..]) {
```

Comment:
Why do we need `!input.is_empty()` ? (and not use `rem` ?)

Reply:
yes, my bad, `!input.is_empty()` is unnecessary, and `rem` is not used, because it's empty for msg_code == 20 (I was trying to use it firstly). Will try to find a problem with `rem`.